lib/ProductOpener/GS1.pm

```diff
@@ -852,6 +852,60 @@ sub assign_field($results_ref, $target_field, $target_value) {
 }
 
 
+sub nutrient_per_quantity ($type, $per, $results_ref, $nid=undef, $nutrient_detail_ref=undef) {
+
+	my $nutrient_field = $nid . $type . "_" . $per;
+
+	my $nutrient_value;
+	my $nutrient_unit;
+
+	# quantityContained may be a single hash, or an array of hashes
+	# e.g. for the energy ENER- field, there are values in kJ and kcal that can be specified in different ways:
+	# - Equadis has 2 ENER- nutrientDetail, each with a single quantityContained hash
+	# - Agena3000 has 1 ENER- nutrientDetail with an array of 2 quantityContained
+	# --> convert a single hash to an array with a hash
+	if ((defined $nutrient_detail_ref->{quantityContained}) and (ref($nutrient_detail_ref->{quantityContained}) ne "ARRAY")) {
+		$nutrient_detail_ref->{quantityContained} = [$nutrient_detail_ref->{quantityContained}];
+	}
+
+	foreach my $quantity_contained_ref (@{$nutrient_detail_ref->{quantityContained}}) {
```

Comment:
For the name of the function, as it is processing the quantityContained field for a nutrient, I would suggest extract_nutrient_quantity_contained()

Reply:
Alright thank you!